experimental/dds/tree2/src/test/feature-libraries/default-field-kinds/optionalChangeRebaser.spec.ts

```diff
@@ -137,29 +176,100 @@ function compose(changes: TaggedChange<OptionalChangeset>[]): OptionalChangeset
 	);
 }
 
-const testChanges: [string, OptionalChangeset][] = [
-	// TODO:AB#4622: This set of edits should be extended to ones with changes to previous content in the field.
-	// If certain types of changes can only be made in some state (e.g. the current format with "wasEmpty"),
-	// we could also consider running multiple exhaustive test suites for meaningfully different starting states.
-	// E.g. in the current format, changes A and B cannot disagree on 'wasEmpty' if they share the same base commit.
-	["SetA", OptionalChange.set("A", false)],
-	["SetB", OptionalChange.set("B", false)],
-	["SetUndefined", OptionalChange.set(undefined, false)],
-	["ChangeChild", OptionalChange.buildChildChange(TestChange.mint([], 1))],
-];
-deepFreeze(testChanges);
+type OptionalFieldTestState = FieldStateTree<string | undefined, OptionalChangeset>;
+
+function* generateChildStates(
+	state: OptionalFieldTestState,
+	tagFromIntention: (intention: number) => RevisionTag,
+	mintIntention: () => number,
+): Iterable<OptionalFieldTestState> {
+	const inputContext = getInputContext(state);
+	if (state.content !== undefined) {
+		const changeChildIntention = mintIntention();
+		yield {
+			content: state.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.buildChildChange(
+						TestChange.mint(inputContext, changeChildIntention),
+					),
+					tagFromIntention(changeChildIntention),
+				),
+				intention: changeChildIntention,
+				description: `ChildChange${changeChildIntention}`,
+			},
+			parent: state,
+		};
+
+		const setUndefinedIntention = mintIntention();
+		yield {
+			content: undefined,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.set(undefined, false),
+					tagFromIntention(setUndefinedIntention),
+				),
+				intention: setUndefinedIntention,
+				description: "Delete",
+			},
+			parent: state,
+		};
+	}
+
+	for (const value of ["A", "B"]) {
+		const setIntention = mintIntention();
+		// Using length of the input context guarantees set operations generated at different times also have different
+		// values, which should tend to be easier to debug
+		const newContents = `${value},${inputContext.length}`;
+		yield {
+			content: newContents,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.set(newContents, state.content === undefined),
+					tagFromIntention(setIntention),
+				),
+				intention: setIntention,
+				description: `Set${value},${inputContext.length}`,
+			},
+			parent: state,
+		};
+	}
+
+	if (state.mostRecentEdit !== undefined) {
+		const undoIntention = mintIntention();
+		yield {
+			content: state.parent?.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					invert(state.mostRecentEdit.changeset),
+					tagFromIntention(undoIntention),
+				),
+				// TODO: Invert/TestChange has some funky logic around negative intentions being used
```

Comment:
It seems like this works as-is, but I'm not too confident it's the most correct thing to do (the call to `invert` does end up inverting the child intention when applicable, which means the "intention" here and the one for at least a portion of the changeset don't match..)

Reply:
The negative intentions are used to help easily identify and cancel-out inverse revisions. It's fine for this layer and `TestChange` no to the same thing on that front so it doesn't matter whether `undoIntention` is `-state.mostRecentEdit.intention` or not.